jpegenc/models/rgb2ycbcr/rgb2ycbcr.py

```diff
@@ -0,0 +1,144 @@
+#!/bin/python
+from myhdl import *
+from commons import *
+from math import *
+
+from math import *
+
+#Fixed point representation with 1 sign bit and 14 fractional bits
+fract_bits = 14
+sign_bits=1
+
+Y_COEFF=[0.2999,0.587,0.114]
+CB_COEFF=[-0.1687,-0.3313,0.5]
+CR_COEFF=[0.5,-0.4187,-0.0813]
+OFFSET=[0,128,128]
+
+Y=[int(round(Y_COEFF[i]*(2**fract_bits )))for i in range(3)]
+Cb=[int(round(CB_COEFF[i]*(2**fract_bits ))) for i in range(3)]
+Cr=[int(round(CR_COEFF[i]*(2**fract_bits ))) for i in range(3)]
+Offset=[int(round(OFFSET[i]*(2**fract_bits ))) for i in range(3)]
+
+class RGB(object):
+
+    def __init__(self, nbits=8):
+        self.nbits=nbits
+        self.red = Signal(intbv(0)[nbits:])
+        self.green = Signal(intbv(0)[nbits:])
+        self.blue = Signal(intbv(0)[nbits:])
+
+    def next(self, r, g, b):
+        self.red.next = r
+        self.green.next = g
+        self.blue.next = b
+
+    def bitLength(self): return self.nbits
+
+
+class YCbCr(object):
+
+    def __init__(self, nbits=8):
+        self.nbits = nbits
+        self.y = Signal(intbv(0)[nbits:])
+        self.cb = Signal(intbv(0)[nbits:])
+        self.cr = Signal(intbv(0)[nbits:])
+
+    def bitLength(self): return self.nbits
+
+
+def rgb2ycbcr(ycbcr, enable_out, rgb, enable_in, clk, reset):
+    """ A RGB to YCbCr converter with reset.
+
+    I/O pins:
+    --------
+    y           : output 8-bit unsigned value in range of 0-127
+    cb          : output 8-bit unsigned value in range of 0-128
+    cr          : output 8-bit unsigned value in range of 0-128
+    enable_out  : output True when output is available
+    r           : input 8-bit unsigned value in range of 0-255
+    g           : input 8-bit unsigned value in range of 0-255
+    b           : input 8-bit unsigned value in range of 0-255
+    enable_in   : input True when input is available
+    clk         : input clock boolean signal
+    reset       : input reset boolean signal
+
+    """
+    
+    #signals for y,cb,cr registers
+    Y_reg,Cb_reg,Cr_reg=[[Signal(intbv(0,-2**(23),2**(23)-1)) for _ in range(3)] for _ in range(3)]
+    Y_sum,Cb_sum,Cr_sum=[Signal(intbv(0,-2**(23),2**(23)-1)) for _ in range(3)]
+    
+    #signals for signed input RGB
+    R_s,G_s,B_s=[Signal(intbv(0,-2**8,2**8-1)) for _ in range(3)]
```

Comment:
I realize the variable names probably match those of the reference design but we want this to be a new "fresh" implementation.  Use the PEP8 guidelines for variables names, etc.

```
red, green, blue = [Signal(intbv(0, min=-colormax, max=colormax)) for _ in range(3)
```